libres/lib/config/conf.cpp

```diff
@@ -1037,8 +1037,17 @@ conf_instance_validate_sub_instances(const conf_instance_type *conf_instance) {
     return ok;
 }
 
-bool conf_instance_get_path_error(const conf_instance_type *conf_instance) {
-    bool path_errors = false;
+/**
+ * This method is internal to the module and cannot be exposed to C via
+ * the headerfile since it returns std::set (which is a C++ type)
+ *
+ * It returns std::set because this is a simple and efficient way to
+ * avoid duplicated keywords in the end-result: Note that the method
+ * recurses and combines results in the last part of the code.
```

Comment:
Maybe call this function `get_path_errors`  ? I think commenting about the export status of this function might get outdated quickly so I think the documentation should be mor localised to what the function does.